test/integration/tests/core.cpp

```diff
@@ -301,6 +301,81 @@ TEST_F(ZNCTest, AwayNotify) {
     client.Write("znc shutdown");
 }
 
+
+TEST_F(ZNCTest, AwayNotifyDetached) {
+    auto znc = Run();
+    auto ircd = ConnectIRCd();
+    auto client = ConnectClient();
```

Comment:
LoginClient() would simplify some of this code. It doesn't send CAP LS, but that's fine, REQ is enough

The test above already tests that away-notify is not advertised by ZNC too early